### PR TITLE
moving ParseSelector to Parse for labels only.

### PR DESCRIPTION
**Pull request by sdminonne**
@smarterclayton @thockin 

In this PR `labels.ParseSelector` is systematically replaced with `labels.Parse`. For the moment I didn't create the `fields` package as agreed in https://github.com/GoogleCloudPlatform/kubernetes/issues/341#issuecomment-75712472

In this PR I've also slightly modified `selector.go` following @smarterclayton suggestion and [this](https://code.google.com/p/go-wiki/wiki/CodeReviewComments).

### CODE REVIEW

**Comment by sdminonne** [comment]
file plugin/pkg/scheduler/factory/factory.go try to parse `DesiredState.Host!=`. Restored the old `ParseSelector` but I'm not sure about. Is this a `field`?

**Comment by bgrant0607** [comment]
Yes, DesiredState.Host is a v1beta1 pod field.

**Comment by sdminonne** [comment]
Thanks

**Comment by brendandburns** [comment]
We use labels all over the place in e2e tests in hack/e2e-suite/*.sh

Can you update the calls there too?

Also, needs rebase.

**Comment by thockin** [comment]
I am not going to have time to review this in the next week.  Brian - who
else can we loop in on this area?

**Comment by bgrant0607** [comment]
@nikhiljindal is working on versioning of field selectors, which is related.

**Comment by sdminonne** [comment]
@nikhiljindal @brendanburns @thockin I've rebased.

I've also modified e2e tests but I couldn't run them  to verify them. (I've no gcloud account) and vagrant env is broken. For the moment I'm updating the new code. Working on my side to run the e2e.

File: pkg/client/request.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ func (r *Request) ParseSelectorParam(paramName, item string) *Request {
 	if r.err != nil {
 		return r
 	}
-	sel, err := labels.ParseSelector(item)
+	sel, err := labels.Parse(item)
```
Comment (nikhiljindal) [inline pkg/client/request.go]: We need to use ParseSelector when paramName = "fields" and Parse when paramName = "labels".

File: hack/e2e-suite/liveness.sh [hunk 2]
```diff
@@ -51,7 +51,7 @@ function teardown() {
 }
 
 function waitForNotPending() {
-  pod_id_list=$(${KUBECTL} get pods -o template '--template={{range.items}}{{.id}} {{end}}' -l test=liveness)
+  pod_id_list=$(${KUBECTL} get pods -o template '--template={{range.items}}{{.id}} {{end}}' -l'test in (liveness)')
```
Comment (nikhiljindal) [inline hack/e2e-suite/liveness.sh]: Why is this change required?
IIUC, both "test=liveness" and "test in (liveness)" will work, right?

**Comment by nikhiljindal** [comment]
Can you please rebase.
We should be able to delete labels.ParseSelector method after rebase. 
And then you can move labels.ParseAndTransformSelector to fields.ParseSelector whenever you plan to do that as per https://github.com/GoogleCloudPlatform/kubernetes/issues/341#issuecomment-75712472

File: hack/e2e-suite/liveness.sh (hunk 2 above)
Comment (sdminonne) [inline hack/e2e-suite/liveness.sh]: yeah. I'll rollback this

**Comment by sdminonne** [comment]
@nikhiljindal working on this

File: pkg/client/request.go (hunk 1 above)
Comment (sdminonne) [inline pkg/client/request.go]: Thanks

**Comment by sdminonne** [comment]
@nikhiljindal rebased. How do you want to proceed to replace ParseSelector with ParseAndTransformSelector?

**Comment by bgrant0607** [comment]
@sdminonne I'd like to get this merged. Did you pick up some other files by mistake, such as hack/e2e-suite/pd.sh?

**Comment by bgrant0607** [comment]
pd.sh was removed:
https://github.com/GoogleCloudPlatform/kubernetes/commit/7bb7ba65774165bcd72acf308420274cf405431f

Please also check hack/e2e-suite/monitoring.sh and hack/lib/test.sh. Those changes don't look relevant to this PR.

Maybe try to rebase again?

When you get this cleaned up, I'd like to merge and iterate from there.

**Comment by sdminonne** [comment]
My bad. I'll fix it asap.
Not  at my desk now.

Dario

@sdminonne <https://github.com/sdminonne> I'd like to get this merged. Did
you pick up some other files by mistake, such as hack/e2e-suite/pd.sh?

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/4835#issuecomment-77422615>
.

**Comment by bgrant0607** [comment]
At some point -- doesn't have to be this PR, please update labels.md with the extended syntax.

**Comment by sdminonne** [comment]
@bgrant0607 PTAL
1) removed hack/e2e-suite/pd.sh
2) hack/e2e-suite/monitoring.sh get rid of spurious modif
3) hack/lib/test.sh needs to be there since hack/test-cmd.sh has been modified with the new syntax and we now have `'-l name in (valid-pod)'` in single quote.

I'll modify labels.md tomorrow

**Comment by bgrant0607** [comment]
LGTM
